modular_skyrat/modules/microfusion/code/phase_emitter.dm

```diff
@@ -0,0 +1,204 @@
+
+	to_chat(user, span_notice("Thermal throttle percent set to: [new_throttle]."))
+
+	if(new_throttle > 100)
+		to_chat(user, span_danger("WARNING: You have input a throttle percentage of more than 100, this may cause emitter damage."))
+
+	throttle_percentage = new_throttle
+
+/obj/item/microfusion_phase_emitter/update_icon_state()
+	. = ..()
+
+	if(damaged)
+		icon_state = "[base_icon_state]_damaged"
+	else
+		switch(get_heat_percent())
+			if(40 to 69)
+				icon_state = "[base_icon_state]_hot"
+			if(70 to INFINITY)
+				icon_state = "[base_icon_state]_critical"
+			else
+				icon_state = base_icon_state
+
+/obj/item/microfusion_phase_emitter/proc/process_fire_delay_and_sound()
+	var/fire_delay_to_add = 0
+	if(integrity < 100)
+		fire_delay_to_add = fire_delay_to_add + (100 - integrity) / 10
+
+	if(current_heat > max_heat)
+		fire_delay_to_add = fire_delay_to_add + (current_heat - max_heat) / 100 //Holy shit this emitter is tanking
+
+	fire_delay = round(fire_delay_to_add, 1.45)
+
+/obj/item/microfusion_phase_emitter/proc/get_heat_icon_state()
+	switch(get_heat_percent())
+		if(40 to 69)
+			return "hot"
+		if(70 to INFINITY)
+			return "critical"
+		else
+			return "normal"
+
+/obj/item/microfusion_phase_emitter/examine(mob/user)
+	. = ..()
+	if(damaged)
+		. += span_danger("It is damaged beyond repair.")
+	else
+		. += span_notice("It has a thermal rating of: [max_heat] C")
+		. += span_notice("It dissipates heat at: [heat_dissipation_per_tick] C")
+		. += span_notice("Heat capacity: [get_heat_percent()]%")
+		. += span_notice("Integrity: [integrity]%")
+		. += span_notice("Thermal throttle: [throttle_percentage]%")
+		. += span_notice("Cooling system: [cooling_system ? "enabled, cooling at [cooling_system_rate] C/s" : "disabled"].")
+
+/obj/item/microfusion_phase_emitter/proc/get_heat_percent()
+	return round(current_heat / max_heat * 100)
+
+/obj/item/microfusion_phase_emitter/proc/check_emitter()
+	if(damaged)
+		return PHASE_FAILURE_DAMAGED
+	if(get_heat_percent() >= throttle_percentage)
+		return PHASE_FAILURE_THROTTLE
+	return SHOT_SUCCESS
+
+/obj/item/microfusion_phase_emitter/proc/add_heat(heat_to_add)
+	current_heat += heat_to_add
+	update_appearance()
+
+/obj/item/microfusion_phase_emitter/proc/kill()
+	damaged = TRUE
+	name = "damaged [name]"
+	playsound(src, 'modular_skyrat/modules/microfusion/sound/overheat.ogg', 70)
+	say("ERROR: Integrity failure!")
+	STOP_PROCESSING(SSobj, src)
+
+/obj/item/microfusion_phase_emitter/enhanced
+	name = "enhanced microfusion phase emitter"
+	desc = "The core of a microfusion projection weapon, produces the laser."
+	max_heat = 1500
+	throttle_percentage = 85
+	heat_dissipation_per_tick = 30
+	cooling_system_rate = 30
+	integrity = 120
+	color = "#ffffcc"
+
+/obj/item/microfusion_phase_emitter/advanced
+	name = "advanced microfusion phase emitter"
+	desc = "The core of a microfusion projection weapon, produces the laser."
```

Comment:
desc = "A third-generation phase emitter, boasting a high capacity for heat, greater dissipation and cooling, and is built using higher-grade materials for more durability."